tests/functional/src/main/java/org/mule/functional/api/classloading/isolation/DependencyResolver.java

```diff
@@ -4,95 +4,212 @@
+                                        List<Dependency> managedDependencies,
+                                        DependencyFilter dependencyFilter) {
+    CollectRequest collectRequest = new CollectRequest();
+    collectRequest.setRoot(root);
+    collectRequest.setDependencies(directDependencies);
+    collectRequest.setManagedDependencies(managedDependencies);
+    collectRequest.setRepositories(Collections.<RemoteRepository>emptyList());
+
+    DependencyNode node;
+    try {
+      node = system.collectDependencies(session, collectRequest).getRoot();
+      DependencyRequest dependencyRequest = new DependencyRequest();
+      dependencyRequest.setRoot(node);
+      dependencyRequest.setCollectRequest(collectRequest);
+      if (dependencyFilter != null) {
+        dependencyRequest.setFilter(dependencyFilter);
       }
-    });
-    return transitiveDependencies;
+
+      node = system.resolveDependencies(session, dependencyRequest).getRoot();
+    } catch (DependencyCollectionException e) {
+      throw new RuntimeException("Error while collecting dependencies", e);
+    } catch (DependencyResolutionException e) {
+      logger.warn(
+                  "Dependencies couldn't be resolved for request '{}', {}",
+                  collectRequest, e.getMessage());
+      node = e.getResult().getRoot();
+      logUnresolvedArtifacts(node, e);
+      throw new RuntimeException("Dependencies couldn't be resolved. See logs to understand which dependencies are missed " +
+          "in the dependency graph", e);
+    }
+
+    List<File> files = getFiles(node);
+    return files;
+  }
+
+  /**
+   * Traverse the {@link DependencyNode} to get the files for each artifact.
+   *
+   * @param node {@link DependencyNode} that represents the dependency graph
+   * @return {@link List} of {@link File}s for each artifact resolved
+   */
+  private List<File> getFiles(DependencyNode node) {
+    PreorderNodeListGenerator nlg = new PreorderNodeListGenerator();
+    node.accept(nlg);
+
+    return nlg.getFiles().stream().map(File::getAbsoluteFile).collect(toList());
   }
 
+  /**
+   * Logs the paths for each dependency not found
+   *
+   * @param node root {@link DependencyNode}, can be a "null" root (imaginary root)
+   * @param e {@link DependencyResolutionException} the error to collect paths.
+   */
+  private void logUnresolvedArtifacts(DependencyNode node, DependencyResolutionException e) {
+    List<ArtifactResult> artifactResults =
+        e.getResult().getArtifactResults().stream().filter(artifactResult -> !artifactResult.getExceptions().isEmpty()).collect(
+                                                                                                                                toList());
+    PathRecordingDependencyVisitor visitor =
+        new PathRecordingDependencyVisitor(new PatternInclusionsDependencyFilter(artifactResults.stream()
+            .map(artifactResult -> toId(
```

Comment:
Please, review all the PR changes and manually fix the formatm this code is pretty un-readeable